src/Symfony/Component/Lock/Store/DbalStore.php

```diff
@@ -0,0 +1,286 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception as DBALException;
+use Doctrine\DBAL\Schema\Schema;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\InvalidTtlException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\PersistingStoreInterface;
+
+/**
+ * DbalStore is a PersistingStoreInterface implementation using a Doctrine DBAL connection.
+ *
+ * Lock metadata are stored in a table. You can use createTable() to initialize
+ * a correctly defined table.
+
+ * CAUTION: This store relies on all client and server nodes to have
+ * synchronized clocks for lock expiry to occur at the correct time.
+ * To ensure locks don't expire prematurely; the TTLs should be set with enough
+ * extra time to account for any clock drift between nodes.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class DbalStore implements PersistingStoreInterface
+{
+    use ExpiringStoreTrait;
+
+    private $conn;
+    private $driver;
+    private $table = 'lock_keys';
+    private $idCol = 'key_id';
+    private $tokenCol = 'key_token';
+    private $expirationCol = 'key_expiration';
+    private $gcProbability;
+    private $initialTtl;
+
+    /**
+     * List of available options:
+     *  * db_table: The name of the table [default: lock_keys]
+     *  * db_id_col: The column where to store the lock key [default: key_id]
+     *  * db_token_col: The column where to store the lock token [default: key_token]
+     *  * db_expiration_col: The column where to store the expiration [default: key_expiration]
+     *
+     * @param Connection $conn          A DBAL Connection instance
+     * @param array      $options       An associative array of options
+     * @param float      $gcProbability Probability expressed as floating number between 0 and 1 to clean old locks
+     * @param int        $initialTtl    The expiration delay of locks in seconds
+     *
+     * @throws InvalidArgumentException When namespace contains invalid characters
+     * @throws InvalidArgumentException When the initial ttl is not valid
+     */
+    public function __construct(Connection $conn, array $options = [], float $gcProbability = 0.01, int $initialTtl = 300)
```

Comment:
`PdoStore` also accepted a DSN. I think, we should keep that feature.

Reply:
Do you think we need this feature in `DbalStore`?

Reply:
If we don't carry that feature over to `DbalStore`, we would effectively remove it with 6.0. And I'm not convinced we should do that.